src/useSubscription/useSubscription.ts

```diff
@@ -9,7 +9,33 @@ import { uniqueValueWatcher } from '@/utilities/uniqueValueWatcher'
 
 const defaultManager = new Manager()
 
-export function useSubscription<T extends Action>(...[action, args, optionsArg = {}]: SubscribeArguments<T>): UseSubscription<T> {
+/**
+ * The `useSubscription` composition manages data sharing across components. Multiple components can subscribe to an `action` (any method or function) and share the response value.
+ *
+ * @example
+ * ```typescript
+ * import { useSubscription } from '@prefecthq/vue-compositions'
+ *
+ * const subscription = useSubscription(action, args, options)
+ * ```
+ *
+ * ## How it works
+ * When creating a subscription for the first time a `Channel` is created. A channel is unique to both the `action` and the `args` used to create the subscription. When a channel is created it immediately executes the action with the given args. If `options.interval` is set the action will execute again at that interval updating the `loading`, `errored`, `error`, and `response` values automatically.
+ *
+ * If another subscription is created with the same action and args, it will not execute the action again. Instead the same channel will be reused. So if multiple components subscribe using the same args the action will only be executed once.
+ *
+ * If any subscriptions have `options.interval` then the shortest interval will be used. All subscriptions will received an updated response at the decided interval, even if a specific subscription has no interval or if it has a greater interval. Interval is basically "maximum age of execution" which defaults to `Infinity`.
+ *
+ * When a subscription is removed the channel interval is recalculated. A subscription can be removed by calling `subscription.unsubscribe()` or the subscription will automatically be removed when the component that created the subscription is unmounted.
```

Comment:
This is really long and also duplicated from the readme. Can we make a short summary? That way we don't have to make sure this and the readme is in sync

Reply:
https://github.com/PrefectHQ/vue-compositions/pull/367/files#r1441095905

Yeah it is duplicated from the README. .. 🤔 

I'm not sure that I think the length is an _issue_ tho. I think it's near the maximum length we probably want but personally I really like how much more accessible this would make this information. Agreed keeping this and the readme in-sync sounds 👎 . 

IMO this is preferable to the README but i wasn't sure about outright deleting it myself. 

All that said tho, I'm more than happy to defer to you on this. Even for a more concise description here - do you have any particular suggestions as you definitely understand this thing better than me 😅 . Or if you'd prefer to keep the README as the source of truth, what do you think about just

```js
/**
 * The `useSubscription` composition manages data sharing across components. Multiple components can subscribe to an `action` (any method or function) and share the response value.
 *
 * @see [README](https://github.com/PrefectHQ/vue-compositions/tree/main/src/useSubscription#readme)
 *
 * @param action - The function to be executed.
 * @param args - Parameters of the action being executed.
 */
```

Reply:
Yeah that snippet preferable to me. I think keeping the readmes is necessary since the whole repository is explained by readmes.

Reply:
Ah I just realized that'd still be duplicating the docs for params/return values since those are inlined over in `susbscription.ts`. Defining them via @param/@return here doesn't work with the extracted return type and also doesn't play as well with destructured return values.

Reply:
Updated in b4a8a53. 👍